service/src/main/kotlin/fi/espoo/evaka/placement/PlacementService.kt

```diff
@@ -644,6 +648,7 @@ LEFT JOIN service_need_option default_sno ON default_sno.default_option AND defa
                 )
             }
             .toList<MissingGroupPlacement>()
+            .filter { it.gap.start <= today.plusMonths(6) }
```

Comment:
On line 618 it checks that the result includes only missing group placements starting from `evakaLaunch`. I think similar check could be introduced for `sixMonthsFromToday`. Like

```
AND daterange(p.start_date, p.end_date, '[]') && daterange(NULL, ${bind(sixMonthsFromToday)}, '[]')
```
Perhaps those two dates could be used to create a single `daterange` as well, i.e. from evakaLaunch until sixMonthsFromToday?

But I agree that doing the filtering in Kotlin is more readable.